packages/cardano-services/src/PgBoss/util.ts

```diff
@@ -0,0 +1,94 @@
+import { DataSource } from 'typeorm';
+import { Logger } from 'ts-log';
+import { Observable, firstValueFrom } from 'rxjs';
+import { Pool } from 'pg';
+import { PoolMetadataEntity, STAKE_POOL_METADATA_QUEUE, StakePoolMetadataTask } from '@cardano-sdk/projection-typeorm';
+import { contextLogger } from '@cardano-sdk/util';
+import { createHttpStakePoolMetadataService } from '../StakePool/HttpStakePoolMetadata/HttpStakePoolMetadataService';
+import PgBoss from 'pg-boss';
+
+const workerQueues = ['pool-metadata'] as const;
+
+export type PgBossQueue = typeof workerQueues[number];
+
+export const isValidQueue = (queue: string): queue is PgBossQueue => workerQueues.includes(queue as PgBossQueue);
+
+type WorkerHandler = (data: unknown) => Promise<void>;
+
+export interface WorkerQueue {
+  createHandler: (dataSource$: Observable<DataSource>, logger: Logger) => WorkerHandler;
+  name: string;
+}
+
+/**
+ * Defines the handler for each pg-boss queue
+ */
+export const queueHandlers: Record<PgBossQueue, WorkerQueue> = {
+  'pool-metadata': {
+    createHandler: (dataSource$, logger) => {
+      const service = createHttpStakePoolMetadataService(logger);
+
+      return (async (task: StakePoolMetadataTask) => {
+        const { metadataJson, poolId, poolRegistrationId } = task;
+        const { hash, url } = metadataJson;
+
+        logger.info('Resolving stake pool metadata...');
+        logger.info({ hash, poolId, poolRegistrationId, url });
+
+        const { errors, metadata } = await service.getStakePoolMetadata(hash, url);
+
+        logger.info('Stake pool metadata resolved');
+        logger.info({ errors, hash, metadata, poolId, poolRegistrationId, url });
+
+        // In case there is some error in fetching extended metadata, the root metadata will be
+        // populated: we need to save it anyway
+        if (metadata) {
+          const dataSource = await firstValueFrom(dataSource$);
+          const repos = dataSource.getRepository(PoolMetadataEntity);
+          const entity = repos.create({
+            ...metadata,
+            hash,
+            poolUpdate: { id: BigInt(poolRegistrationId) },
+            stakePool: { id: poolId }
+          });
+
+          await repos.save(entity);
+        }
+
+        // TODO: Store the error in a dedicated table
+        // Ref: LW-6409
+
+        // In case of errors the handler must throw in order to let pg-boss to retry the job
+        if (errors?.length) throw errors[0];
```

Comment:
2b90ae0f0 should solve this

Reply:
I think the 1st part is required for initial implementation:
> Need to add some logic on when to retry the job. For example, if it successfully fetched the metadata but the hash or ext signature doesn't match then it will get exactly same result on retry - no need to do that.

Reply:
48d52c32f should solve this